packages/fx-core/src/component/resource/botService/messages.ts

```diff
@@ -3,34 +3,27 @@
 
 import { getDefaultString, getLocalizedString } from "../../../common/localizeUtils";
 
+export function localString(key: string, ...params: any[]): [string, string] {
+  return [getDefaultString(key, ...params), getLocalizedString(key, ...params)];
+}
 export class Messages {
-  public static readonly SomethingIsMissing = (something: string): [string, string] => [
-    getDefaultString("plugins.bot.SomethingIsMissing", something),
-    getLocalizedString("plugins.bot.SomethingIsMissing", something),
-  ];
-
-  public static readonly FailToProvisionSomeResource = (resource: string): [string, string] => [
-    getDefaultString("plugins.bot.FailedToProvision", resource),
-    getLocalizedString("plugins.bot.FailedToProvision", resource),
-  ];
-  public static readonly FailToUpdateConfigs = (something: string): [string, string] => [
-    getDefaultString("plugins.bot.FailedToUpdateConfigs", something),
-    getLocalizedString("plugins.bot.FailedToUpdateConfigs", something),
-  ];
-  public static readonly BotRegistrationNotFoundWith = (botId: string): [string, string] => [
-    getDefaultString("plugins.bot.BotRegistrationNotFoundWith", botId),
-    getLocalizedString("plugins.bot.BotRegistrationNotFoundWith", botId),
-  ];
-  public static readonly FailToUpdateMessageEndpoint = (endpoint: string): [string, string] => [
-    getDefaultString("plugins.bot.FailedUpdateMessageEndpoint", endpoint),
-    getLocalizedString("plugins.bot.FailedUpdateMessageEndpoint", endpoint),
-  ];
-
-  public static readonly FailToCallAppStudioForCheckingAADApp = [
-    getDefaultString("plugins.bot.FailToCallAppStudioApi"),
-    getLocalizedString("plugins.bot.FailToCallAppStudioApi"),
-  ];
+  public static readonly SomethingIsMissing = (something: string): [string, string] =>
```

Comment:
Maybe there is no need to declare the type here, just let TS deduce it automatically.

Reply:
It turned out to be type of `string[]` not two strings `[string, string]`.